### ISSUE TITLE
About SpatialSoftMax in SAD

### ISSUE THREAD

**Issue by khw321** [cardwing/Codes-for-Lane-Detection#235]
@cardwing In the SAD module, the shape of feature map after power and sum is B x H/8 x W/8,
and I have test the SpatialSoftMax op is computed in the batch size dimension. If the batch size per GPU is 1, the elements in the attention map of block 3 and block 4 will equal to 1. If I misunderstand the SAD code?
`             (1): nn.Power     -- B x C x H/8 x W/8
             (2): nn.Sum         -- B x H/8 x W/8
             (3): nn.SpatialUpSamplingBilinear(8)  -- B x H x W
             (4): nn.SpatialSoftMax  -- B x H x W,  compute in B dimension
`

**Comment by cardwing** [comment]
The SpatialSoftMax is performed in the spatial dimension (HxW) not the batch dimension.

**Comment by khw321** [comment]
But I have test the code in the ENet-Label-Torch, and the SpatialSoftMax is performed in the batch dimension. I have another question

**Comment by khw321** [comment]
But I have test the code in the ENet-Label-Torch, the SpatialSoftMax is performed in the batch dimension. 
And I have another question about performed in the spatial dimension. After Power and Sum, the output feature map's value is between 0 to hundreds. So if use SpatialSoftMax in the spatial dimension, the output will have only one element close to 1, and the others close to 0.

**Comment by cardwing** [comment]
According to the description in the official site, [spatialsoftmax](https://github.com/torch/nn/blob/master/doc/transfer.md#spatialsoftmax) is performed on the spatial dimension (-1 and -2 dim). After using spatialsoftmax, many elements in the attention map will indeed close to 0 and only several elements will have relatively big values.

**Comment by khw321** [comment]
Yes, but in practice, the output feature map's value is between 0 to hundreds, after softmax the biggest element will greater than 0.99, and others is very close to 0. Can the SAD loss work well? And in the ENet-Label-Torch code, spatialsoftmax is performed in the batch dimension. If the SAD code is wrong?

**Comment by Evolto-Z** [comment]
You asked the question I'm confused with for a long time, and I completely agree with your opinions! I think, maybe, SAD can replace SpatialSoftmax with Sigmoid to do the normalization.
